/*!
* @file
* @brief ISession
*
* @copyright Copyright (C) University of Bristol 2017
*    This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
*    If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
*    See LICENSE file for details.
* @date 01/02/2018
* @author Richard Collins <richard.collins@bristol.ac.uk>
*/

syntax = "proto3";
import public "google/protobuf/empty.proto";
import public "Messages.proto";

package cqp.remote;

/**
 * @brief The ISift class
 * Interface for the communication between Sifting classes
 */
service ISession {

    /**
    * @brief SessionStarting
    * Called by the other session controller, instructing the controller to start
    * generating key
    * @param request Session details to start the session
    * @return status
    */
    rpc SessionStarting(SessionDetails) returns (google.protobuf.Empty);

    /**
    * The key generation process should be stopped
    * @return status
    */
    rpc SessionEnding(google.protobuf.Empty) returns (google.protobuf.Empty);

}
